### PR TITLE
6.3

### PR DESCRIPTION
**Pull request by kat3chrome**

### CODE REVIEW

File: semester_1/hw_6/6.3/main.c [hunk 1]
```diff
@@ -0,0 +1,79 @@
+#include <stdio.h>
+#include <stdbool.h>
+#include "set.h"
+
+int main()
```
Comment (jzuken) [inline semester_1/hw_6/6.3/main.c]: definitely lost: 8 bytes in 1 blocks

File: semester_1/hw_6/6.3/main.c [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    {
+      case 1:
+      {
+        int value = 0;
+        printf("Enter value: ");
+        scanf("%d", &value);
+        addElement(value, set);
+        break;
+      }
+      case 2:
```
Comment (jzuken) [inline semester_1/hw_6/6.3/main.c]: Enter number of command: 6
( 5 ( 3 ( 2 null null )( 4 null null ))( 8 ( 7 null null )( 9 null null )))
Enter number of command: 2
Enter value: 3
Enter number of command: 6
( 5 ( 3 ( 2 null null ) null )( 8 ( 7 null null )( 9 null null )))

File: semester_1/hw_6/6.3/set.c [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+      elementWithMinimemValue->parent->rightChild = NULL;
+    }
+  }
+
+  int value = elementWithMinimemValue->value;
+  free(elementWithMinimemValue);
+  return value;
+}
+
+void removeElement(int value, Set* set)
```
Comment (jzuken) [inline semester_1/hw_6/6.3/set.c]: длинновата функция

File: semester_1/hw_6/6.3/set.c [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+}
+
+bool isInTheSet(int value, Set* set)
+{
+  if (findSetElement(value, set) != NULL)
+  {
+    return true;
+  }
+
+  return false;
```
Comment (SpirinEgor) [inline semester_1/hw_6/6.3/set.c]: Можно упростить

**Pull request merged**